### ISSUE TITLE
Syntax highlighting stop working after loading certain python files

### ISSUE THREAD

**Issue by AlmostSeagull** [microsoft/pylance-release#3886]
Type: <b>Bug</b>

# Behaviour: 
Text color does not change
## Expected vs. Actual

Expected: 
Several words would get several colors. 
Actual: 
Text stays white (except strings and comments, but that seems to be a different extension's work). The text on no other file will change color until a reload of any kind has been done AND the originally affected python file is not opened (opening it triggers the bug)

## Steps to reproduce:

1. I don't know what caused it, so step 1 will be unknown
2. Open an affected file (I have no clue as of the logic of it, it's not bigger than others, it has correct syntax, not all other files in the folder are affected)
3. Text stays white when it shouldn't (am open to help figure out what caused it and how to fix. There's probably a bad file somewhere interfering with the extension)

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.4
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Global
-   Value of the `python.languageServer` setting: Default

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

```
XXX
```

</p>
</details>

<details>

<summary>User Settings</summary>

<p>

```

languageServer: "Pylance"

```

</p>
</details>

Extension version: 2022.20.2
VS Code version: Code 1.74.3 (97dec172d3256f8ca4bfb2143f3f76b503ca0534, 2023-01-09T16:59:02.252Z)
OS version: Windows_NT x64 10.0.19042
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 2700X Eight-Core Processor          (16 x 3693)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.95GB (24.92GB free)|
|Process Argv|--crash-reporter-id bb989480-520b-4040-875d-aee2338cc0df|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
azure-dev_surveyone:30548225
vsccc:30610678
pyindex848cf:30577861
nodejswelcome1:30587005
3biah626:30602489
pyind779:30647091
f6dab269:30613381
pythonsymbol12:30652348
2i9eh265:30646982

```

</details>

**Comment by AlmostSeagull** [comment]
Hey, thanks for getting back to me, I have created a log for you guys. The problem file from my initial description is "ItemRando.py". This log details me turning the log on, opening a file that works (Logic.py if I remember correctly), ItemRando.py, which bricks the color highlighting and some other files that it then also fails to highlight. 

[Info  - 11:21:11] (8424) Pylance language server 2023.1.40 (pyright e96b3ece) starting
[Info  - 11:21:11] (8424) Server root directory: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist
[Info  - 11:21:11] (8424) Starting service instance "DK64-Randomizer"
[Info  - 11:21:11] (8424) Auto-indent enabled
[Info  - 11:21:11] (8424) No configuration file found.
[Info  - 11:21:11] (8424) No pyproject.toml file found.
[Info  - 11:21:11] (8424) Setting pythonPath for service "DK64-Randomizer": "C:\Users\Walter\AppData\Local\Programs\Python\Python310\python.exe"
[Warn  - 11:21:11] (8424) stubPath g:\DK64R\DK64-Randomizer\typings is not a valid directory.
[Info  - 11:21:11] (8424) Assuming Python version 3.10
[Info  - 11:21:11] (8424) Assuming Python platform Windows
[Info  - 11:21:11] (8424) Search paths for g:\DK64R\DK64-Randomizer
[Info  - 11:21:11] (8424)   c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib
[Info  - 11:21:11] (8424)   g:\DK64R\DK64-Randomizer
[Info  - 11:21:11] (8424)   g:\DK64R\DK64-Randomizer\typings
[Info  - 11:21:11] (8424)   c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stubs\...
[Info  - 11:21:11] (8424)   c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\bundled\stubs
[Info  - 11:21:11] (8424)   c:\Users\Walter\AppData\Local\Programs\Python\Python310\DLLs
[Info  - 11:21:11] (8424)   c:\Users\Walter\AppData\Local\Programs\Python\Python310\Lib
[Info  - 11:21:11] (8424)   c:\Users\Walter\AppData\Local\Programs\Python\Python310
[Info  - 11:21:11] (8424)   c:\Users\Walter\AppData\Local\Programs\Python\Python310\Lib\site-packages
[Info  - 11:21:11] (8424) Adding fs watcher for library directories:
 C:\Users\Walter\AppData\Local\Programs\Python\Python310\DLLs
C:\Users\Walter\AppData\Local\Programs\Python\Python310\lib
C:\Users\Walter\AppData\Local\Programs\Python\Python310
C:\Users\Walter\AppData\Local\Programs\Python\Python310\lib\site-packages
[Info  - 11:21:11] (8424) Adding fs watcher for directories:
 g:\DK64R\DK64-Randomizer
[Info  - 11:21:11] (8424) Searching for source files
[Info  - 11:21:11] (8424) Found 204 source files
[Info  - 11:21:11] (8424) Background analysis(1) root directory: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist
[Info  - 11:21:11] (8424) Background analysis(1) started
(8424) Background analysis message: setConfigOptions
(8424) Background analysis message: setImportResolver
(8424) Background analysis message: ensurePartialStubPackages
(8424) Background analysis message: setTrackedFiles
(8424) Background analysis message: markAllFilesDirty
(8424) Background analysis message: analyze
(8424) [IDX(FG)] index libraries g:\DK64R\DK64-Randomizer (index) ...
(8424) [IDX(FG)]   read stdlib indices (35ms)
(8424) [IDX(FG)] index libraries g:\DK64R\DK64-Randomizer (index) [succeed] (38ms)
(8424) Background analysis message: setFileOpened
(8424) [FG] parsing: g:\DK64R\DK64-Randomizer\randomizer\Logic.py (116ms)
(8424) [FG] parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\builtins.pyi [fs read 1ms] (89ms)
(8424) [FG] binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\builtins.pyi (43ms)
(8424) [FG] binding: g:\DK64R\DK64-Randomizer\randomizer\Logic.py (20ms)
(8424) Background analysis message: analyze
(8424) [BG(1)] analyzing: g:\DK64R\DK64-Randomizer\randomizer\Logic.py ...
(8424) [BG(1)]   parsing: g:\DK64R\DK64-Randomizer\randomizer\Logic.py (109ms)
(8424) [BG(1)]   parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\builtins.pyi [fs read 1ms] (77ms)
(8424) [BG(1)]   binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\builtins.pyi (33ms)
(8424) [BG(1)]   binding: g:\DK64R\DK64-Randomizer\randomizer\Logic.py (17ms)
(8424) [BG(1)]   checking: g:\DK64R\DK64-Randomizer\randomizer\Logic.py ...
(8424) [BG(1)]     parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\typing.pyi [fs read 2ms] (22ms)
(8424) [BG(1)]     binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\typing.pyi (9ms)
(8424) [BG(1)]     parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\typing_extensions.pyi [fs read 0ms] (3ms)
(8424) [BG(1)]     binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\typing_extensions.pyi (1ms)
(8424) [BG(1)]     parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi [fs read 1ms] (9ms)
(8424) [BG(1)]     binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi (2ms)
(8424) [BG(1)]     parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\types.pyi [fs read 1ms] (8ms)
(8424) [BG(1)]     binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\types.pyi (4ms)
(8424) [BG(1)]     parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\abc.pyi [fs read 0ms] (1ms)
(8424) [BG(1)]     binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\abc.pyi (2ms)
(8424) [BG(1)]     parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\math.pyi [fs read 0ms] (3ms)
(8424) [BG(1)]     binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\math.pyi (1ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\AngryAztec.py [fs read 0ms] (5ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\AngryAztec.py (6ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\CreepyCastle.py [fs read 0ms] (7ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\CreepyCastle.py (2ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\CrystalCaves.py [fs read 0ms] (10ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\CrystalCaves.py (1ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\DKIsles.py [fs read 1ms] (1ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\DKIsles.py (1ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\FranticFactory.py [fs read 0ms] (3ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\FranticFactory.py (1ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\FungiForest.py [fs read 0ms] (4ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\FungiForest.py (1ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\GloomyGalleon.py [fs read 0ms] (3ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\GloomyGalleon.py (1ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\JungleJapes.py [fs read 1ms] (4ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\CollectibleLogicFiles\JungleJapes.py (1ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\AngryAztec.py [fs read 0ms] (5ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\AngryAztec.py (2ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\CreepyCastle.py [fs read 1ms] (5ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\CreepyCastle.py (2ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\CrystalCaves.py [fs read 1ms] (5ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\CrystalCaves.py (3ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\DKIsles.py [fs read 0ms] (4ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\DKIsles.py (3ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\FranticFactory.py [fs read 0ms] (6ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\FranticFactory.py (1ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\FungiForest.py [fs read 1ms] (5ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\FungiForest.py (4ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\GloomyGalleon.py [fs read 0ms] (5ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\GloomyGalleon.py (4ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\HideoutHelm.py [fs read 0ms] (2ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\HideoutHelm.py (2ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\JungleJapes.py [fs read 0ms] (5ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\JungleJapes.py (2ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\Shops.py [fs read 0ms] (4ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\Shops.py (13ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\Locations.py [fs read 0ms] (7ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\Locations.py (7ms)
(8424) [BG(1)]     parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\enum.pyi [fs read 0ms] (4ms)
(8424) [BG(1)]     binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\enum.pyi (1ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\Regions.py [fs read 0ms] (2ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\Regions.py (2ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\Types.py [fs read 0ms] (1ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\Types.py (1ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Lists\Item.py [fs read 1ms] (9ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Lists\Item.py (2ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\Items.py [fs read 0ms] (2ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\Items.py (1ms)
(8424) [BG(1)]     parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\collections\abc.pyi [fs read 0ms] (0ms)
(8424) [BG(1)]     binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\collections\abc.pyi ...
(8424) [BG(1)]       parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\_collections_abc.pyi [fs read 1ms] (1ms)
(8424) [BG(1)]       binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\_collections_abc.pyi (0ms)
(8424) [BG(1)]     binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\collections\abc.pyi (3ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\Kongs.py [fs read 0ms] (1ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\Kongs.py (0ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\MoveTypes.py [fs read 1ms] (1ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\MoveTypes.py (0ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\Levels.py [fs read 0ms] (1ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\Levels.py (0ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\Collectibles.py [fs read 0ms] (1ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\Collectibles.py (0ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\Events.py [fs read 0ms] (1ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\Events.py (0ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\Time.py [fs read 0ms] (0ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\Time.py (1ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Lists\Location.py [fs read 0ms] (21ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Lists\Location.py (6ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Lists\MapsAndExits.py [fs read 1ms] (4ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Lists\MapsAndExits.py (2ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\VendorType.py [fs read 1ms] (1ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\VendorType.py (0ms)
(8424) IntelliCode model c:\Users\Walter\.vscode\extensions\visualstudioexptteam.vscodeintellicode-1.2.30\dist\bundledModels\python_intellisense-members-lstm-pylance
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Lists\ShufflableExit.py [fs read 1ms] (11ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Lists\ShufflableExit.py (4ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Prices.py [fs read 1ms] (5ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Prices.py (2ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Lists\LevelInfo.py [fs read 0ms] (0ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Lists\LevelInfo.py (1ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\Transitions.py [fs read 1ms] (2ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\Transitions.py (2ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\LogicClasses.py [fs read 0ms] (2ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\LogicClasses.py (1ms)
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\ExitCategories.py [fs read 0ms] (0ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\ExitCategories.py (1ms)
(8424) Loading ONNX runtime...
(8424) Loaded ONNX runtime. Creating IntelliCode session...
2023-01-31 11:22:31.033149 [I:onnxruntime:, inference_session.cc:263 operator()] Flush-to-zero and denormal-as-zero are off
2023-01-31 11:22:31.034150 [I:onnxruntime:, inference_session.cc:271 ConstructorCommon] Creating and using per session threadpools since use_per_session_threads_ is true
2023-01-31 11:22:31.034335 [I:onnxruntime:, inference_session.cc:292 ConstructorCommon] Dynamic block base set to 0
2023-01-31 11:22:31.056144 [I:onnxruntime:, inference_session.cc:1222 Initialize] Initializing session.
2023-01-31 11:22:31.056260 [I:onnxruntime:, inference_session.cc:1259 Initialize] Adding default CPU execution provider.
2023-01-31 11:22:31.058776 [I:onnxruntime:, reshape_fusion.cc:42 ApplyImpl] Total fused reshape node count: 0
2023-01-31 11:22:31.059893 [I:onnxruntime:, graph.cc:3553 CleanUnusedInitializersAndNodeArgs] Removing NodeArg 'rnn/transpose'. It is no longer used by any node.
2023-01-31 11:22:31.060004 [I:onnxruntime:, graph.cc:3553 CleanUnusedInitializersAndNodeArgs] Removing NodeArg 'Gather_out0'. It is no longer used by any node.
2023-01-31 11:22:31.060089 [I:onnxruntime:, graph.cc:3553 CleanUnusedInitializersAndNodeArgs] Removing NodeArg 'Transpose_out0'. It is no longer used by any node.
2023-01-31 11:22:31.060705 [I:onnxruntime:, reshape_fusion.cc:42 ApplyImpl] Total fused reshape node count: 0
2023-01-31 11:22:31.068028 [V:onnxruntime:, session_state.cc:1010 VerifyEachNodeIsAssignedToAnEp] Node placements
2023-01-31 11:22:31.068184 [V:onnxruntime:, session_state.cc:1013 VerifyEachNodeIsAssignedToAnEp]  All nodes placed on [CPUExecutionProvider]. Number of nodes: 59
2023-01-31 11:22:31.068663 [V:onnxruntime:, session_state.cc:66 CreateGraphInfo] SaveMLValueNameIndexMapping
2023-01-31 11:22:31.068785 [V:onnxruntime:, session_state.cc:112 CreateGraphInfo] Done saving OrtValue mappings.
2023-01-31 11:22:31.069154 [I:onnxruntime:, session_state_utils.cc:199 SaveInitializedTensors] Saving initialized tensors.
2023-01-31 11:22:31.087519 [I:onnxruntime:, session_state_utils.cc:342 SaveInitializedTensors] Done saving initialized tensors
2023-01-31 11:22:31.088396 [I:onnxruntime:, inference_session.cc:1488 Initialize] Session successfully initialized.
(8424) Created IntelliCode session.
(8424) Initialize deeplearning succeeded
(8424) [BG(1)]     parsing: g:\DK64R\DK64-Randomizer\randomizer\Enums\MinigameType.py [fs read 0ms] (1ms)
(8424) [BG(1)]     binding: g:\DK64R\DK64-Randomizer\randomizer\Enums\MinigameType.py (0ms)
(8424) [BG(1)]   checking: g:\DK64R\DK64-Randomizer\randomizer\Logic.py (4153ms)
[Info  - 11:22:32] (8424) [BG(1)] Long operation: checking: g:\DK64R\DK64-Randomizer\randomizer\Logic.py (4153ms)
(8424) [BG(1)] analyzing: g:\DK64R\DK64-Randomizer\randomizer\Logic.py (4393ms)
[Info  - 11:22:32] (8424) [BG(1)] Long operation: analyzing: g:\DK64R\DK64-Randomizer\randomizer\Logic.py (4393ms)
(8424) Background analysis message: getDiagnosticsForRange
(8424) Background analysis message: getSemanticTokens full
(8424) [BG(1)] getSemanticTokens full at g:\DK64R\DK64-Randomizer\randomizer\Logic.py (110ms)
(8424) Background analysis message: getSemanticTokens range
(8424) [BG(1)] getSemanticTokens range 0:0 - 117:0 at g:\DK64R\DK64-Randomizer\randomizer\Logic.py (12ms)
(8424) Background analysis message: getDiagnosticsForRange
(8424) Background analysis message: getDiagnosticsForRange
(8424) Background analysis message: getDiagnosticsForRange
(8424) Background analysis message: getDiagnosticsForRange
(8424) Background analysis message: resumeAnalysis
(8424) Background analysis message: getDiagnosticsForRange
(8424) Background analysis message: getDiagnosticsForRange
(8424) Background analysis message: getDiagnosticsForRange
(8424) Background analysis message: getDiagnosticsForRange
(8424) Background analysis message: setFileOpened
(8424) [FG] parsing: g:\DK64R\DK64-Randomizer\randomizer\Patching\ItemRando.py (36ms)
(8424) [FG] binding: g:\DK64R\DK64-Randomizer\randomizer\Patching\ItemRando.py (8ms)
(8424) Background analysis message: getDiagnosticsForRange
(8424) Background analysis message: getSemanticTokens full
(8424) [BG(1)] getSemanticTokens full at g:\DK64R\DK64-Randomizer\randomizer\Patching\ItemRando.py ...
(8424) [BG(1)]   parsing: g:\DK64R\DK64-Randomizer\randomizer\Patching\ItemRando.py (10ms)
(8424) [BG(1)]   binding: g:\DK64R\DK64-Randomizer\randomizer\Patching\ItemRando.py (4ms)
(8424) [BG(1)]   parsing: g:\DK64R\DK64-Randomizer\js.py [fs read 1ms] (1ms)
(8424) [BG(1)]   binding: g:\DK64R\DK64-Randomizer\js.py (1ms)
(8424) [BG(1)]   parsing: g:\DK64R\DK64-Randomizer\randomizer\Patching\Patcher.py [fs read 1ms] (3ms)
(8424) [BG(1)]   binding: g:\DK64R\DK64-Randomizer\randomizer\Patching\Patcher.py (1ms)
(8424) [BG(1)]   parsing: g:\DK64R\DK64-Randomizer\randomizer\Spoiler.py [fs read 0ms] (21ms)
(8424) [BG(1)]   binding: g:\DK64R\DK64-Randomizer\randomizer\Spoiler.py (10ms)
(8424) [BG(1)]   parsing: g:\DK64R\DK64-Randomizer\randomizer\Patching\Lib.py [fs read 1ms] (15ms)
(8424) [BG(1)]   binding: g:\DK64R\DK64-Randomizer\randomizer\Patching\Lib.py (2ms)
(8424) [BG(1)]   parsing: g:\DK64R\DK64-Randomizer\randomizer\Settings.py [fs read 0ms] (20ms)
(8424) [BG(1)]   binding: g:\DK64R\DK64-Randomizer\randomizer\Settings.py (12ms)
(8424) [BG(1)]   parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\array.pyi [fs read 0ms] (2ms)
(8424) [BG(1)]   binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\array.pyi (2ms)
(8424) [BG(1)]   parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\mmap.pyi [fs read 0ms] (2ms)
(8424) [BG(1)]   binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\mmap.pyi (1ms)
(8424) [BG(1)]   parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\ctypes\__init__.pyi [fs read 0ms] (7ms)
(8424) [BG(1)]   binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\ctypes\__init__.pyi (4ms)
(8424) [BG(1)]   parsing: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\pickle.pyi [fs read 1ms] (3ms)
(8424) [BG(1)]   binding: c:\Users\Walter\.vscode\extensions\ms-python.vscode-pylance-2023.1.40\dist\typeshed-fallback\stdlib\pickle.pyi (1ms)
(8424) [FG] parsing: g:\DK64R\DK64-Randomizer\randomizer\Lists\Patches.py (28ms)
(8424) [FG] binding: g:\DK64R\DK64-Randomizer\randomizer\Lists\Patches.py (10ms)
(8424) [FG] parsing: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\JungleJapes.py (13ms)
(8424) [FG] binding: g:\DK64R\DK64-Randomizer\randomizer\LogicFiles\JungleJapes.py (5ms)

**Comment by AlmostSeagull** [comment]
> from the description, my gut feeling is we are hitting some perf issue when opening the file you mentioned causing us to take long time to process the request and that makes all other requests to be blocked. for this particular case, if we can get the file that is causing the issue, it will probably most helpful for us to pinpoint the root cause.

Okay, I read that way too late. Here is a link to the file in question: https://github.com/2dos/DK64-Randomizer/blob/dev/randomizer/Patching/ItemRando.py

Not sure what you mean by perf issue. My PC is fine, so I'd guess performance is possibly limited by something else. Does sound like it should be just a matter of waiting, for as far as it doesn't get stuck in an infinite loop somewhere or something similar. Thanks again for the reply, and looking forward to hearing from you or someone else :)

**Comment by erictraut** [comment]
@AlmostSeagull, thanks for the repro steps. I was able to replicate the behavior you're seeing, and I introduced a fix that addresses the problem. The fix will be included in this week's insider's build of pylance and in next week's production release.

**Comment by AlmostSeagull** [comment]
Awesome to hear! Thanks a lot! So ehm... this isn't my code, but ehm... is there anything more constructive I can tell the person whose code this is? Not that what you said is bad, but I don't see myself poking him and saying "oh btw this person said you may want to consider refactoring this method" 
Also kinda guessing it's an optional thing to refactor, given the upcoming fix you mentioned, but I don't mean to miss out on solid advice.

**Comment by bschnurr** [comment]
This issue has been fixed in prerelease version 2023.2.11, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023211-1-february-2023-prerelease)
